### ISSUE TITLE
Property 'anime' does not exist on type 'JakanMisc'.

### ISSUE THREAD

**Issue by SRK7Kyros** [Lamarcke/Jakan#2]
![image](https://user-images.githubusercontent.com/107617546/208297998-3ceec84c-6ce4-41c4-b77f-c0d14f214a9b.png)

It weirdly says the error for some reason... it's actually just the example usage tho.
What did i miss?

**Comment by Lamarcke** [comment]
Hey, it seems that most editors have a hard time figuring out the return type with this builder pattern. 

You can circumvent this by specifying the client type in the `withXXXX` methods, like this:

```typescript
import { Jakan, JakanSearch } from "jakan"

const api = new Jakan().forSearch().withMemory<JakanSearch>()

...

``` 

Your editor should now understand that `api` is of type `JakanSearch`. 
This also applies to `withWebStorage`, `withMemory` and `withRedis`.

You can also import and use `JakanUsers` and `JakanMisc` as types.

I'm still trying to figure a way to make this detection automatic. If anyone has a good suggestion on how to achieve this (without having to use different methods for each builder type), please let me know :)

**Comment by Lamarcke** [comment]
Just for a quick explanation:
I didn't want to, but i will need to change the API for the `Jakan` class in `1.0.0`.
However, this will make it possible to avoid complex Typescript-only assertions. 

This will only be available when `1.0.0` is out in the NPM registry. I still want to finish `JakanMisc` #5 before this.

**Comment by Lamarcke** [comment]
Closed by #9
